examples/uart/src/uart.cc

```diff
@@ -0,0 +1,78 @@
+
+
+#include "ns_core.h"
+#include "ns_ambiqsuite_harness.h"
+#include "ns_peripherals_power.h"
+#include "ns_timer.h"
+#include "am_hal_mcuctrl.h"
+#include "ns_uart.h"
+#include <stdio.h>
+#include <string.h>
+#include "ns_peripherals_button.h"
+
+#if defined(AM_PART_APOLLO3) || defined(AM_PART_APOLLO3P)
+#define RX_BUFFER_SIZE 256
+#define TX_BUFFER_SIZE 256
+static uint8_t g_pui8TxBuffer[TX_BUFFER_SIZE];
+static uint8_t g_pui8RxBuffer[RX_BUFFER_SIZE];
+#endif
+
+// UART0 interrupt handler.
+am_hal_uart_config_t g_sUartConfig =
```

Comment:
All of this should be 'hidden' behind the ns_uart API (generally, ns examples should have minimal ambiqsuite code).